### PR TITLE
Fix typos in KeyUsage constant names

### PR DESCRIPTION
**Pull request by abursavich**
**What type of PR is this?**
/kind api-change
/kind cleanup

**What this PR does / why we need it**:
This fixes typos in KeyUsage constant names.

I am generating code from the API packages and this tickled an edge case where I'm trying to generate nicer enum names.

**Which issue(s) this PR fixes**:
N/A

**Special notes for your reviewer**:
As possible precedence for changing constant names, one of these typos was introduced by a PR attempting to fix typos: https://github.com/kubernetes/kubernetes/pull/59464

**Does this PR introduce a user-facing change?**:
I'm not 100% sure what constitutes a "user-facing" change... This **could** break users of the Go `k8s.io/api/certificates/v1beta1` package.

```release-note
Fix typos in `certificates.k8s.io/v1beta1` KeyUsage constant names: `UsageContentCommittment` becomes `UsageContentCommitment` and `UsageNetscapSGC` becomes `UsageNetscapeSGC`.
```

### CODE REVIEW

**Review by draveness — COMMENTED** [review]
/ok-to-test

Thanks for the catch

**Comment by neolit123** [comment]
> I'm not 100% sure what constitutes a "user-facing" change... This could break users of the Go k8s.io/api/certificates/v1beta1 package.

changes in kubeadm or `pkg/apis/certificates/types.go` are OK, since these packages should not be vendored, however `staging/src/k8s.io/api/certificates/v1beta1/types.go` could break users.

i think the correct approach is to introduce these changes in a new `certificates/v1beta2` version.

**Comment by neolit123** [comment]
@abursavich 
you can ask in the k8s slack channel #sig-auth about this.

**Comment by abursavich** [comment]
> i think the correct approach is to introduce these changes in a new `certificates/v1beta2` version.

If the change would break encoding (rpc, storage, etc.), then I don't think there would be a question -- v1beta2 would be required. But this change only breaks the Go package API and, rightly or wrongly, there does seem to be plenty of precedence for that.

> you can ask in the k8s slack channel #sig-auth about this.

I dropped the question in the slack channel and will wait for a response... Thanks!

**Comment by deads2k** [comment]
> > I'm not 100% sure what constitutes a "user-facing" change... This could break users of the Go k8s.io/api/certificates/v1beta1 package.
> 
> changes in kubeadm or `pkg/apis/certificates/types.go` are OK, since these packages should not be vendored, however `staging/src/k8s.io/api/certificates/v1beta1/types.go` could break users.
> 
> i think the correct approach is to introduce these changes in a new `certificates/v1beta2` version.

We do not guarantee source compatibility between releases.  This came up on the sig-arch mailing list too.  Source incompatible changes like this are allowed without new packages.

**Comment by abursavich** [comment]
I didn't try to grok all 85k lines of log output, but it seems like those test failures were unrelated and around copying test artifacts after the fact.

/retest

**Review by draveness — COMMENTED** [review]
/lgtm

Thanks, Please update the release note with the original and replacement constants.

**Comment by abursavich** [comment]
> Thanks, Please update the release note with the original and replacement constants.

Done, thanks.

**Comment by abursavich** [comment]
Gentle nudge... small change waiting 2 weeks for approval.

**Comment by neolit123** [comment]
i can only approve the kubeadm change. the rest is on the api-approvers.
/approve

File: pkg/apis/certificates/types.go [hunk 1]
```diff
@@ -118,27 +118,27 @@ type CertificateSigningRequestList struct {
...
-	UsageIPsecEndSystem     KeyUsage = "ipsec end system"
-	UsageIPsecTunnel        KeyUsage = "ipsec tunnel"
-	UsageIPsecUser          KeyUsage = "ipsec user"
-	UsageTimestamping       KeyUsage = "timestamping"
-	UsageOCSPSigning        KeyUsage = "ocsp signing"
-	UsageMicrosoftSGC       KeyUsage = "microsoft sgc"
-	UsageNetscapSGC         KeyUsage = "netscape sgc"
+	UsageSigning           KeyUsage = "signing"
+	UsageDigitalSignature  KeyUsage = "digital signature"
+	UsageContentCommitment KeyUsage = "content commitment"
```
Comment (liggitt) [inline pkg/apis/certificates/types.go]: this also exists in staging/src/k8s.io/api/certificates/v1beta1/types.go

**Pull request merged**